### PR TITLE
Add Qt6 support

### PR DESCRIPTION
**Pull request by lkc-rc**
Here is the promised support for Qt6, sorry it took so wanted, I wanted to get the changes reviewed on our side first ^^

Since I've written my version, I saw you also added some code to install the xkb files from meson. I've left my version in the PR, because I find it more concise, but it's up to you whether you want to keep this change.

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -1,17 +1,34 @@
 project('qfreerdp', 'cpp',
   version : '1.0.0',
-  meson_version : '>= 1.0',
+  meson_version : '>= 0.63',  # before version 0.63.0 Meson would fail to find Qt 6.1 or later
```
Comment (hardening) [inline meson.build:3]: I had troubles with the version shipped in Ubuntu 22.04 with `qt_deps.get_variable('QMAKE_SPEC')` it's 0.61 AFAICT, do you think `0.63` behaves better ? 
TBH that made me remove the distro meson and update it to the most recent version on all my test platforms and I've set `1.0` as what seemed like a safe value...

File: meson.build [hunk 2]
```diff
@@ -29,5 +46,16 @@ xcursor_deps = dependency('xcursor')
...
 with_tests = get_option('tests').enabled()
 
+
+# Windows specific keyboard layouts
+xkb_data_path = get_option('xkb_data_path')
+if xkb_data_path == ''
+    xkb_data_path = join_paths(get_option('prefix'), get_option('datadir'), 'X11', 'xkb')
+endif
+
+install_subdir('xkb',
```
Comment (hardening) [inline meson.build:56]: Oh you're right that's so much simpler with `install_subdir`

File: meson.build (hunk 1 above)
Comment (lkc-rc) [reply meson.build:3]: We had the same issues, in fact, all of the testing was done on meson 1.3.2 installed from pip. But I didn't want to restrict your users if, according to meson's docs, 0.63 is enough: https://mesonbuild.com/Qt6-module.html

Comment (hardening) [reply meson.build:3]: Ok let's take that and we'll see if people complains

**Review by hardening — APPROVED** [review]
Tested and works great with Qt6. Thank you for the contribution.

**Comment by hardening** [comment]
I've seen a segfault that sometimes happens with the main application from the Qt example directory when you open the "About window" and close it. But I guess it's not related to this PR.

**Pull request merged**